services/user.py

```diff
@@ -0,0 +1,52 @@
+from django.contrib.auth import get_user_model
+
+
+User = get_user_model()
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: str = None,
+    first_name: str = None,
+    last_name: str = None,
+) -> User:
+    user = User.objects.create_user(username=username)
+    user.set_password(password)
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+
+    user.save()
+    return user
+
+
+def get_user(user_id: int) -> User:
+    return User.objects.get(pk=user_id)
+
+
+def update_user(
+    user_id: int,
+    username: str = None,
+    password: str = None,
+    email: str = None,
+    first_name: str = None,
+    last_name: str = None,
+) -> User:
+    user = User.objects.get(pk=user_id)
+    if username:
+        user.username = username
+    if password:
+        user.set_password(password)
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+
+    user.save()
```

Comment:
In the `update_user` function, you can also reduce the number of `if` checks by using the `update` method of the `QuerySet` API. This method allows you to update certain fields of the model directly in the database, without having to load it into Python. This can be more efficient if you are updating a lot of instances at once because it only requires one database query. However, keep in mind that the `update` method does not call the `save` method or trigger the `pre_save` or `post_save` signals, so if you need those to happen, you should stick with your current approach.